spiffworkflow-frontend/src/routes/ProcessInstanceShow.tsx

```diff
@@ -1099,71 +1028,114 @@ export default function ProcessInstanceShow({ variant }: OwnProps) {
       params.process_model_id ? params.process_model_id : ''
     );
 
+    const getTabs = () => {
```

Comment:
any time you call `ability.can`, it can suffer from the problem that the permissions are not loaded yet and then it says you don't have access but you actually do, so it's safest to wrap the call in a check to ensure that permissionsLoaded is true. i have added one to line 1026 in ProcessInstanceShow and it didn't obviously break anything. this check is not required when using the jsx `<Can>` component for some reason, which is why we have a bias for that construct, but it's not always possible and it can definitely be a lot more natural to avoid it.

Reply:
AH - that makes a ton of sense - thank you for the explanation.